src/storage/v3/shard_rsm.cpp

```diff
@@ -112,6 +124,248 @@ Value ConstructValueVertex(const memgraph::storage::v3::VertexAccessor &acc, mem
+      break;
+    }
+  }
+  return Value{};
+}
+
+std::vector<Value> ConvertToValueVectorFromTypedValueVector(std::vector<memgraph::storage::v3::TypedValue> &&vec) {
+  std::vector<Value> ret;
+  ret.reserve(vec.size());
+  for (auto &&elem : vec) {
+    ret.emplace_back(FromTypedValueToValue(std::move(elem)));
+  }
+  return ret;
+}
+
+std::vector<PropertyId> NamesToProperties(const std::vector<std::string> &property_names,
+                                          memgraph::expr::DbAccessor &dba) {
+  std::vector<PropertyId> properties;
+  properties.reserve(property_names.size());
+  for (const auto &name : property_names) {
+    properties.push_back(dba.NameToProperty(name));
+  }
+  return properties;
+}
+
+std::vector<memgraph::storage::v3::LabelId> NamesToLabels(const std::vector<std::string> &label_names,
+                                                          memgraph::expr::DbAccessor &dba) {
+  std::vector<memgraph::storage::v3::LabelId> labels;
+  labels.reserve(label_names.size());
+  for (const auto &name : label_names) {
+    labels.push_back(dba.NameToLabel(name));
+  }
+  return labels;
+}
+
+template <class TExpression>
+auto Eval(TExpression *expr, memgraph::expr::EvaluationContext ctx, memgraph::expr::AstStorage &storage,
+          memgraph::storage::v3::ExpressionEvaluator &eval, memgraph::expr::DbAccessor &dba) {
+  ctx.properties = NamesToProperties(storage.properties_, dba);
+  ctx.labels = NamesToLabels(storage.labels_, dba);
+  auto value = expr->Accept(eval);
+  return value;
+}
+
+std::vector<PropertyId> GetPropertiesFromAcessor(
+    const std::map<memgraph::storage::v3::PropertyId, memgraph::storage::v3::PropertyValue> &properties) {
+  std::vector<PropertyId> ret_properties;
+  ret_properties.reserve(properties.size());
+
+  for (const auto &prop : properties) {
+    ret_properties.emplace_back(prop.first);
+  }
+
+  return ret_properties;
+}
+
+memgraph::expr::EvaluationContext CreateEvaluationContext(
+    const std::optional<memgraph::storage::v3::VertexAccessor> &v_acc,
+    const std::optional<memgraph::storage::v3::EdgeAccessor> &e_acc) {
+  memgraph::expr::EvaluationContext ctx;
+
+  std::vector<PropertyId> vertex_props;
+  std::vector<PropertyId> edge_props;
+  std::vector<memgraph::storage::v3::LabelId> labels_in_shard;
+
+  if (v_acc) {
+    const auto labels = v_acc->Labels(View::OLD);
+    labels_in_shard.reserve(labels.GetValue().size());
+
+    for (const auto &label : labels.GetValue()) {
+      labels_in_shard.emplace_back(label);
+    }
+
+    vertex_props = GetPropertiesFromAcessor(v_acc->Properties(View::OLD).GetValue());
+  }
+  if (e_acc) {
+    edge_props = GetPropertiesFromAcessor(e_acc->Properties(View::OLD).GetValue());
+  }
+
+  std::vector<PropertyId> all_properties(vertex_props.size() + edge_props.size());
+
+  // Sorting might not needed, since the props are being converted from std::map
+  std::sort(vertex_props.begin(), vertex_props.end());
```

Comment:
General question: do we need to specify a sorting lambda or the default is good enough?

Reply:
Yes, a lambda should have been there, I was not sure if the sorting was needed there at all or not, but I will just remove the sorting from there, since the elements should in a sorted order at that point anyways.